src/features/modal/Modal.tsx

```diff
@@ -0,0 +1,30 @@
+import { PropsWithChildren } from "react";
+import Portal from "../../components/@common/Portal/Portal";
+import * as S from "./Modal.styles";
+
+type Props = PropsWithChildren<{
+  open: boolean;
+  onClose: () => void;
+}>;
+
+const Modal = ({ open, onClose, children }: Props) => {
+  const stopPropagation = (e: React.MouseEvent<HTMLDivElement>) => {
+    e.stopPropagation();
+  };
+
+  return (
+    <>
+      {open && (
+        <Portal>
+          <S.ModalBackdrop onClick={onClose}>
+            <S.ModalContainer onClick={stopPropagation}>
+              <S.ModalContent>{children}</S.ModalContent>
+            </S.ModalContainer>
+          </S.ModalBackdrop>
+        </Portal>
+      )}
+    </>
```

Comment:
이런 식으로 작성하는 방법도 있을거 같네요(취향)

```suggestion
  if (!open) return null;

  return (
        <Portal>
          <S.ModalBackdrop onClick={onClose}>
            <S.ModalContainer onClick={stopPropagation}>
              <S.ModalContent>{children}</S.ModalContent>
            </S.ModalContainer>
          </S.ModalBackdrop>
        </Portal>
       )
```

Reply:
오오 맞습니다! early return 방식이 불필요한 Fragment 같은거 사용하지 않아도 돼서 훨씬 깔끔하고, 인덴트 뎁스도 줄어서 가독성이 더 좋아진 것 같습니다.
프리코스 때는 요구사항 때문에 익숙해져서 거의 기본 소양 수준으로 익숙 했었는데, 막상 오랜만에 하다 보니 까먹고 있었던 게 좀 민망하네요.. 상기시켜주셔서 감사합니다 ㅎㅎ
https://github.com/woowacourse/react-shopping-cart/pull/375/commits/4fd331b9c111679e5ec06e40acda27388c199788